shamir_mnemonic/__init__.py

```diff
@@ -596,6 +597,25 @@ def generate_mnemonics(
     ]
 
 
+def generate_mnemonics(
+    group_threshold, groups, master_secret, passphrase=b"", iteration_exponent=0
+):
```

Comment:
Why does this function take `master_secret` instead of `encrypted_master_secret`? Is it to allow the user to select `master_secret`? Then the function name should clearly say so, e.g. `generate_mnemonics_from_master_secret` or similar.

Having one function `generate_mnemonics_from_data` that takes EMS and another one `generate_mnemonics` that takes MS, is IMHO very confusing and is exactly what caused the 2.1.3 fiasco.

Reply:
I totally agree that there is no way to tell what generate_mnemonics_from_data() and generate_mnemonics() do without reading the documentation, which is a step backwards. The reason why I didn't have a generate_mnemonics_from_data() function in my implementation was because I couldn't think of a nice name and I wanted to avoid confusion between the two functions. 

As for the naming of the functions, my main requirement is that the names match the terminology in the specification. Secondly I would prefer if we could maintain the high level API, because there are already several other implementations of SLIP-0039 which have adapted this API into different programming languages and it's really unfair to change it in the reference implementation now.